src/ZkFold/Symbolic/Data/Maybe.hs

```diff
@@ -48,3 +50,16 @@ instance SymbolicData a (u (ArithmeticCircuit a))
     restore (h:ts) = Maybe h (restore ts)
     restore _      = Haskell.error "restore ArithmeticCircuit: wrong number of arguments"
     typeSize = 1 + typeSize @a @(u (ArithmeticCircuit a))
+
+maybe :: forall a b f .
+    Conditional (Bool a) b =>
+    DiscreteField (Bool a) a =>
+    b -> (f a -> b) -> Maybe f a -> b
+maybe d h x@(Maybe _ v) = bool @(Bool a) d (h v) $ isNothing x
```

Comment:
Moving it to `ZkFold.Symbolic.Data.Maybe` would be better.

Reply:
@vlasin But it's in `Maybe` module already. Or did you mean another one?